assets/scripts/dialogs/Analytics/MiniGraph.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import _ from 'lodash'
+
+import {
+  XYPlot,
+  XAxis,
+  YAxis,
+  VerticalGridLines,
+  HorizontalGridLines,
+  VerticalBarSeries
+} from 'react-vis'
+
+const totalCount = (arr, dataKeys) => {
+  if (!arr || !arr.length) return
+  const sums = {}
+
+  _.each(arr, function (item) {
+    _.each(dataKeys, function (option) {
+      sums[option] = (sums[option] || 0) + item[option]
+    })
+  })
+  return _.toPairs(sums).map(item => {
+    const [ key, value ] = item
```

Comment:
Can we use native ES6 methods here? e.g. [`Array.prototype.forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach), [`Array.prototype.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map), [`Object.entries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries)

Reply:
I've been unclear on this, sorry! You shouldn't need to call the prototype function itself. I was only using these as a reference to the original method. Would this work?

```js
arr.forEach((item) => {
  dataKeys.forEach((option) => {
    sums[option] = (sums[option] || 0) + item[option]
  })
})
```

On further consideration, we probably could do better than this, with `.map()` and `reduce()`, instead of setting `const sums = {}`?